From: Michael S. Tsirkin <mst@redhat.com>
Subject: [PATCH] vhost: error handling fix
References: FATE#311977
Git-commit: 615cc2211c17ed05a2a5d94abdac6c340a8ea508
Patch-mainline: v2.6.36

vhost should set worker to NULL on cgroups attach failure,
so that we won't try to destroy the worker again on close.

Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 drivers/vhost/vhost.c |    1 +
 1 file changed, 1 insertion(+)

Index: b/drivers/vhost/vhost.c
===================================================================
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -298,6 +298,7 @@ static long vhost_dev_set_owner(struct v
 	return 0;
 err_cgroup:
 	kthread_stop(worker);
+	dev->worker = NULL;
 err_worker:
 	if (dev->mm)
 		mmput(dev->mm);
